clicommand/agent_start.go

```diff
@@ -94,6 +94,7 @@ type AgentStartConfig struct {
 	NoPlugins                   bool     `cli:"no-plugins"`
 	NoPluginValidation          bool     `cli:"no-plugin-validation"`
 	NoPTY                       bool     `cli:"no-pty"`
+	NoTelemetry                 bool     `cli:"no-telemetry"`
```

Comment:
I thin "no telemetry" is too broad. Folks might expect us not to send operating system or process information under this name.

I'd suggest instead we use `--no-features-reporting` to be more specific.